src/backend/joanie/core/serializers.py

```diff
@@ -1,71 +1,334 @@
             "end",
             "enrollment_start",
             "enrollment_end",
             "state",
-            "position",
         ]
 
-    @staticmethod
-    def get_position(obj):
-        """Get position of course run linked for a product"""
-        return obj.course_run.course.product_relations.get(
-            product=obj.order.product
-        ).position
+
+class OrderLiteSerializer(serializers.ModelSerializer):
+    """
+    Minimal Order model serializer
+    """
+
+    id = serializers.CharField(read_only=True, source="uid")
+    price = serializers.DecimalField(
+        coerce_to_string=False,
+        decimal_places=2,
+        max_digits=9,
+        min_value=0,
+        read_only=True,
+    )
+    enrollments = CourseRunEnrollmentSerializer(
+        many=True, read_only=True, required=False
+    )
+    product = serializers.SlugRelatedField(read_only=True, slug_field="uid")
+
+    class Meta:
+        model = models.Order
+        fields = [
+            "id",
+            "created_on",
+            "price",
+            "enrollments",
+            "product",
+            "state",
+        ]
+        read_only_fields = [
+            "id",
+            "created_on",
+            "price",
+            "enrollments",
+            "product",
+            "state",
+        ]
+
+
+class CourseSerializer(serializers.ModelSerializer):
+    """
+    Serialize all information about a course.
+    """
+
+    organization = OrganizationSerializer(read_only=True)
+    products = ProductSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = models.Course
+        fields = [
+            "code",
+            "organization",
+            "title",
+            "products",
+        ]
+        read_only_fields = [
+            "code",
+            "organization",
+            "title",
+            "products",
+        ]
+
+    def get_orders(self, instance):
+        """
+        If an user is authenticated, retrieves its orders related to the serializer Course instance
+        else return None
+        """
+        try:
+            username = self.context["username"]
+            orders = (
+                models.Order.objects.filter(
+                    owner__username=username,
+                    course=instance,
+                    state__in=[
+                        enums.ORDER_STATE_FAILED,
+                        enums.ORDER_STATE_FINISHED,
+                        enums.ORDER_STATE_PAID,
+                        enums.ORDER_STATE_PENDING,
+                    ],
+                )
+                .select_related("product")
+                .prefetch_related("enrollments__course_run")
+            )
+
+            return OrderLiteSerializer(orders, many=True).data
+        except KeyError:
+            return None
+
+    def to_representation(self, instance):
+        """
+        When user is not authenticated, we return the cached representation,
+        otherwise we generate representation as well
```

Comment:
This is outdated now
```suggestion
        Cache the serializer representation that does not vary from user to user
       then, if user is authenticated, add private information to the representatio
```